client/transport/transport.go

```diff
@@ -0,0 +1,53 @@
+package transport
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-es-indexer-go/core"
+	"github.com/multiversx/mx-chain-es-indexer-go/core/request"
+	"github.com/multiversx/mx-chain-es-indexer-go/metrics"
+)
+
+type metricsTransport struct {
+	statusMetrics core.StatusMetricsHandler
+	transport     http.RoundTripper
+}
+
+// NewMetricsTransport will create a new instance of metricsTransport
+func NewMetricsTransport(statusMetrics core.StatusMetricsHandler) (*metricsTransport, error) {
+	if check.IfNil(statusMetrics) {
+		return nil, core.ErrNilMetricsHandler
+	}
+
+	return &metricsTransport{
+		statusMetrics: statusMetrics,
+		transport:     http.DefaultTransport,
+	}, nil
+}
+
+// RoundTrip implements the http.RoundTripper interface and is used as a wrapper around the underlying
+// transport to collect and record metrics related to the HTTP request/response cycle.
+func (m *metricsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	startTime := time.Now()
+	size := req.ContentLength
+	resp, err := m.transport.RoundTrip(req)
+	duration := time.Since(startTime)
+
+	valueFromCtx := req.Context().Value(request.ContextKey)
+	if valueFromCtx == nil {
+		return resp, err
```

Comment:
add a test with nil value on ctx for 100% coverage?

Reply:
added